Source/WebCore/loader/FrameLoader.cpp

```diff
@@ -2570,6 +2572,7 @@ CachePolicy FrameLoader::subresourceCachePolicy(const URL& url) const
 void FrameLoader::dispatchDidFailProvisionalLoad(DocumentLoader& provisionalDocumentLoader, const ResourceError& error, WillInternallyHandleFailure willInternallyHandleFailure)
 {
     m_provisionalLoadErrorBeingHandledURL = provisionalDocumentLoader.url();
+    m_errorOccurredInLoading = true;
```

Comment:
This seems to only handle provisional load failures, which is the only thing tested.  We need to also cover and test the case where a load fails after it has committed.

TEST(WKDownload, FailAndResume) has such a failure.  It has the server respond with Content-Length: 10000 then sends 5k then terminates.  It may more reliably commit if it uses a content-length of 10mb and sends 5mb then terminates.

Reply:
`m_errorOccurredInLoading = true` is also set in `FrameLoader::checkLoadCompleteForThisFrame`

```
        auto& error = documentLoader->mainDocumentError();

        AXObjectCache::AXLoadingEvent loadingEvent;
        if (!error.isNull()) {
            FRAMELOADER_RELEASE_LOG(ResourceLoading, "checkLoadCompleteForThisFrame: Finished frame load with error (isTimeout = %d, isCancellation = %d, errorCode = %d)", error.isTimeout(), error.isCancellation(), error.errorCode());
            m_client->dispatchDidFailLoad(error);
            loadingEvent = AXObjectCache::AXLoadingFailed;
            m_errorOccurredInLoading = true;
        } else {
            FRAMELOADER_RELEASE_LOG(ResourceLoading, "checkLoadCompleteForThisFrame: Finished frame load");
#if ENABLE(DATA_DETECTION)
            RefPtr document = m_frame->document();
            OptionSet<DataDetectorType> types = m_frame->settings().dataDetectorTypes();
            if (document && types) {
                auto documentLevelResults = retainPtr(DataDetection::detectContentInRange(makeRangeSelectingNodeContents(*document), types, m_client->dataDetectionReferenceDate()));
                m_frame->dataDetectionResults().setDocumentLevelResults(documentLevelResults.get());
                if (m_frame->isMainFrame())
                    m_client->dispatchDidFinishDataDetection(documentLevelResults.get());
            }
#endif
            m_client->dispatchDidFinishLoad();
            loadingEvent = AXObjectCache::AXLoadingFinished;
        }
```

Right before dispatchDidFailLoad is called. I believe that would cover the case where the load fails after it has committed.

I do agree that I need a test for that though.

Reply:
I added the test for this.